lib/streamlit/runtime/secrets.py

```diff
@@ -170,26 +177,39 @@ def _parse(self, print_exceptions: bool) -> Mapping[str, Any]:
             if self._secrets is not None:
                 return self._secrets
 
-            try:
-                with open(self._file_path, encoding="utf-8") as f:
-                    secrets_file_str = f.read()
-            except FileNotFoundError:
-                if print_exceptions:
-                    st.error(f"Secrets file not found. Expected at: {self._file_path}")
-                raise
-
-            try:
-                secrets = toml.loads(secrets_file_str)
-            except:
+            # It's fine for a user to only have one secrets.toml file defined, so
+            # we ignore individual FileNotFoundErrors when attempting to read files
+            # below and only raise an exception if we weren't able read *any* secrets
+            # file.
+            found_secrets_file = False
+            secrets = {}
+
+            for path in self._file_paths:
+                try:
+                    with open(path, encoding="utf-8") as f:
+                        secrets_file_str = f.read()
+                    found_secrets_file = True
+                except FileNotFoundError:
+                    continue
+
+                try:
+                    secrets.update(toml.loads(secrets_file_str))
+                except:
+                    if print_exceptions:
+                        st.error(f"Error parsing secrets file at {path}")
+                    raise
+
+            if not found_secrets_file:
+                err_msg = f"No secrets files found. Valid paths for a secrets.toml file are: {self._file_paths}"
```

Comment:
btw @sfc-gh-jcarroll: I didn't put much thought into the exact error message wording here, so feel free to rewrite this as you see fit

Reply:
I'll do a pass on it (and anything similar in the PR) later this week. TY!

Reply:
is it more or less pythonic to print the list literal (as currently) vs something like `{', '.join(secrets_file_locations)}` which gives an output like:

> No secrets files found. Valid paths for a secrets.toml file are: /Users/jcarroll/.streamlit/secrets.toml, /Users/jcarroll/python/streamlit/st-explore/.streamlit/secrets.toml

It could also be quoted but just remove the brackets. Whichever you think is better. Otherwise I think the text is great, and the rest of the PR looks great to me in terms of developer interactions.

Reply:
Not sure which is more Pythonic, I mostly just printed the list literal because it was easier 😆 

I think dropping the brackets looks nicer, though, so I'll change the message to do that